### PR TITLE
#146 Write the deployed yaml resources to separate yaml files

### LINKED ISSUE #146: Write the deployed yaml resources to files when doing 'make deploy' target
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**
Optional. A clear and concise description of any alternative solutions or features you've considered.

**Additional context**
Optional. Add any other context or screenshots about the feature request here.

### LINKED ISSUE #183: cluster-wide scaledown problem: extra permissions needed and log panic
**Describe the bug**
When operator watches all namespaces and perform scale down some permissions are missing
and it will cause log panic as it passes non-string type of key to log.Info() when logging events.

**To Reproduce**
1. Get X version of the source
2. '...'

**Platform (please complete the following information):**
 - OS: [e.g. Fedora]
 - Version [e.g. 32]
 - Go [e.g. 1.13.4

**Expected behavior**
Optional, use if not obvious. A clear and concise description of what you expected to happen.

**Screenshots**
Optional, add screenshots to help explain your problem.

**Additional context**
Optional. Add any other context about the problem here.

### PR DESCRIPTION
**Pull request by gaohoward**
Also:
added missing permissions for doing scaledown cluster wide
added eventLog function to drainer controller to handle the event log
instead of using log.Info() which requires keys to always be string type

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 resources:
-- webhook_secret.yaml
```
Comment (RoddieKieley) [inline config/manager/kustomization.yaml:2]: @gaohoward This is likely a result of the ENABLE_WEBHOOKS work as it gets dynamically added and removed based on the value assigned to ENABLE_WEBHOOKS. Probably doesn't matter if it's included or not; with the default being false maybe it's good to have it not set for the moment.

Comment (gaohoward) [reply config/manager/kustomization.yaml:2]: yeah I think that's the result of the ENABLE_WEBHOOK flag. I'll restore it.

File: pkg/draincontroller/controller.go [hunk 2]
```diff
@@ -118,6 +120,11 @@ type Controller struct {
 	client client.Client
 }
 
+func eventLog(format string, args ...interface{}) {
```
[the remark this answers is not in the document]
Comment (gaohoward) [reply pkg/draincontroller/controller.go:123]: This change (as well the adding permissions to controller) is not part of the main purpose. I think I may split them into another PR. This changes are related to when a watch-all operator scale down brokers - it needs some extra permissions to do so and also can cause panic here:
https://github.com/artemiscloud/activemq-artemis-operator/blob/fc62ff29759dbb10a5fe9d41444318c98f4512d9/pkg/draincontroller/controller.go#L602

Note the **sts**  passed in will eventually passed into this eventLog's args. If we use log.Info() method like here
https://github.com/artemiscloud/activemq-artemis-operator/blob/fc62ff29759dbb10a5fe9d41444318c98f4512d9/pkg/draincontroller/controller.go#L143
then the log.Info() requires all keys to be of type string, so you'll get panic like this:

```
2022-03-28T14:01:20.428Z	DPANIC	controller_v1beta1activemqartemisscaledown	odd number of arguments passed as key-value pairs for logging	{"ignored key": [{"kind":"StatefulSet","namespace":"howard","name":"ex-aao-ss","uid":"b7567ca0-cf99-4746-a91e-641bdc69f580","apiVersion":"apps/v1","resourceVersion":"1624346"},"Normal","DrainSuccess","drain Pod ex-aao-ss-1 in StatefulSet ex-aao-ss completed successfully"]}
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartLogging.func1
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:282
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:308
E0328 14:01:20.428545       1 runtime.go:76] Observed a panic: odd number of arguments passed as key-value pairs for logging
goroutine 597 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1973540, 0xc0008823f0)
	/go/pkg/mod/k8s.io/apimachinery@v0.22.2/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.22.2/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1973540, 0xc0008823f0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0004e60c0, 0xc000b502c0, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.19.0/zapcore/entry.go:232 +0x532
go.uber.org/zap.(*Logger).DPanic(0xc00017e5a0, 0x1cd6a0a, 0x3d, 0xc000b502c0, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.19.0/logger.go:220 +0x85
github.com/go-logr/zapr.handleFields(0xc00017e5a0, 0xc000882330, 0x1, 0x1, 0x0, 0x0, 0x0, 0x7f0eb2827b20, 0x8, 0x18)
	/go/pkg/mod/github.com/go-logr/zapr@v0.4.0/zapr.go:100 +0x5e5
github.com/go-logr/zapr.(*zapLogger).Info(0xc000419840, 0x1ca46c5, 0x26, 0xc000882330, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.4.0/zapr.go:127 +0xb0
sigs.k8s.io/controller-runtime/pkg/log.(*DelegatingLogger).Info(0xc00041d8f0, 0x1ca46c5, 0x26, 0xc000882330, 0x1, 0x1)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.0/pkg/log/deleg.go:130 +0xa2
github.com/artemiscloud/activemq-artemis-operator/pkg/draincontroller.eventLog(0x1ca46c5, 0x26, 0xc000b501c0, 0x4, 0x4)
	/workspace/pkg/draincontroller/controller.go:128 +0x1b5
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartLogging.func1(0xc0000ae500)
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:282 +0x193
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x1eb8d08, 0xc0009468d0, 0xc00049e4c0)
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:308 +0xef
created by k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:299 +0x6e
panic: odd number of arguments passed as key-value pairs for logging [recovered]
	panic: odd number of arguments passed as key-value pairs for logging

goroutine 597 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.22.2/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1973540, 0xc0008823f0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0004e60c0, 0xc000b502c0, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.19.0/zapcore/entry.go:232 +0x532
go.uber.org/zap.(*Logger).DPanic(0xc00017e5a0, 0x1cd6a0a, 0x3d, 0xc000b502c0, 0x1, 0x1)
	/go/pkg/mod/go.uber.org/zap@v1.19.0/logger.go:220 +0x85
github.com/go-logr/zapr.handleFields(0xc00017e5a0, 0xc000882330, 0x1, 0x1, 0x0, 0x0, 0x0, 0x7f0eb2827b20, 0x8, 0x18)
	/go/pkg/mod/github.com/go-logr/zapr@v0.4.0/zapr.go:100 +0x5e5
github.com/go-logr/zapr.(*zapLogger).Info(0xc000419840, 0x1ca46c5, 0x26, 0xc000882330, 0x1, 0x1)
	/go/pkg/mod/github.com/go-logr/zapr@v0.4.0/zapr.go:127 +0xb0
sigs.k8s.io/controller-runtime/pkg/log.(*DelegatingLogger).Info(0xc00041d8f0, 0x1ca46c5, 0x26, 0xc000882330, 0x1, 0x1)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.0/pkg/log/deleg.go:130 +0xa2
github.com/artemiscloud/activemq-artemis-operator/pkg/draincontroller.eventLog(0x1ca46c5, 0x26, 0xc000b501c0, 0x4, 0x4)
	/workspace/pkg/draincontroller/controller.go:128 +0x1b5
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartLogging.func1(0xc0000ae500)
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:282 +0x193
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x1eb8d08, 0xc0009468d0, 0xc00049e4c0)
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:308 +0xef
created by k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher
	/go/pkg/mod/k8s.io/client-go@v0.22.2/tools/record/event.go:299 +0x6e
```
causing operator to restart.

[the remark this answers is not in the document]
Comment (RoddieKieley) [reply pkg/draincontroller/controller.go:123]: Yeah I think this is related in so far as the required permissions need a home in the yaml, but having it separated with the notes above would be beneficial and more clear.

**Comment by gaohoward** [comment]
@RoddieKieley Hi Roddie I've update the branch. I splitted the PR into 2 commits.

**Pull request merged**